### PR TITLE
feat: add builder babel doc

### PR DESCRIPTION
**Pull request by sanyuan0704**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/builder/builder-doc/zh/config/tools/babel.md [hunk 1]
```diff
@@ -1 +1,184 @@
-TODO
+- Type: Object | Function
```
Comment (chenjiahan) [inline packages/builder/builder-doc/zh/config/tools/babel.md:1]: 建议加个引号

Type: Object | Function -> Type: `Object | Function`

File: packages/builder/builder-doc/zh/config/tools/babel.md [hunk 2]
```diff
@@ -1 +1,184 @@
...
+
+- Default: `undefined`
```
Comment (chenjiahan) [inline packages/builder/builder-doc/zh/config/tools/babel.md:3]: Default 和 Type 都是简单类型的话，中间可以不空出一行

File: packages/builder/builder-doc/zh/config/tools/babel.md [hunk 3]
```diff
@@ -1 +1,184 @@
...
+
+通过 `tools.babel` 可以修改 [babel-loader](https://github.com/babel/babel-loader) 的配置项
```
Comment (chenjiahan) [inline packages/builder/builder-doc/zh/config/tools/babel.md:5]: 配置项后面加个句号~

File: packages/builder/builder-doc/zh/config/tools/babel.md [hunk 4]
```diff
@@ -1 +1,184 @@
...
+
+### 类型
```
Comment (chenjiahan) [inline packages/builder/builder-doc/zh/config/tools/babel.md:7]: 共享出去的代码片段不建议包含 h3 标题哈

File: packages/builder/builder-doc/en/config/tools/babel.md [hunk 5]
```diff
@@ -1 +1,184 @@
...
+};
+```
+
+## Util Functions
+
+When `tools.babel` is of type Function, the tool functions available for the second parameter are as follows:
+
+### addPlugins
+
+Add Babel plugin.For exapmle:
```
Comment (chenjiahan) [inline packages/builder/builder-doc/en/config/tools/babel.md:61]: For exapmle 前面加个空格

File: packages/builder/builder-doc/zh/config/tools/babel.md [hunk 6]
```diff
@@ -1 +1,184 @@
...
+```
+
+#### addIncludes
+
+默认情况下 Babel 只会编译 src 目录下的业务代码，使用 addIncludes 你可以指定 Babel 编译 node_modules 下的一些文件。比如:
+
+```js
+export default {
+  tools: {
+    babel(config, { addInclude }) {
```
Comment (chenjiahan) [inline packages/builder/builder-doc/zh/config/tools/babel.md:129]: addInclude -> addIncludes

File: packages/builder/builder-doc/zh/config/tools/babel.md [hunk 7]
```diff
@@ -1 +1,184 @@
...
+#### addExcludes
+
+和 `addIncludes` 相反，指定 Babel 编译时排除某些文件。
+
+比如不编译 `src/example` 目录下的文件:
+
+```js
+export default {
+  tools: {
+    babel(config, { addExclude }) {
```
Comment (chenjiahan) [inline packages/builder/builder-doc/zh/config/tools/babel.md:145]: addExclude -> addExcludes

File: packages/builder/builder-doc/zh/config/tools/babel.md (hunk 4 above)
Comment (sanyuan0704) [reply packages/builder/builder-doc/zh/config/tools/babel.md:7]: 这对用户就不太友好了呀，右边的侧边栏就少锚点了

File: packages/builder/builder-doc/en/config/tools/babel.md (hunk 5 above)
Comment (sanyuan0704) [reply packages/builder/builder-doc/en/config/tools/babel.md:61]: 可

File: packages/builder/builder-doc/zh/config/tools/babel.md [hunk 8]
```diff
@@ -1 +1,184 @@
...
+      addExcludes('src/example');
+    },
+  },
+};
+```
+
+#### modifyPresetEnvOptions
+
+修改 `@babel/preset-env` 的配置项，比如:
```
Comment (chenjiahan) [inline packages/builder/builder-doc/zh/config/tools/babel.md:154]: modify 是与默认值 merge，还是 replace 默认值，文档里可以提一下

File: packages/builder/builder-doc/zh/config/tools/babel.md (hunk 4 above)
Comment (chenjiahan) [reply packages/builder/builder-doc/zh/config/tools/babel.md:7]: 这个锚点感觉没啥必要嘞，不是有效信息

File: packages/builder/builder-doc/zh/config/tools/babel.md (hunk 7 above)
Comment (sanyuan0704) [reply packages/builder/builder-doc/zh/config/tools/babel.md:145]: 我统一更正下

File: packages/builder/builder-doc/zh/config/tools/babel.md (hunk 4 above)
Comment (sanyuan0704) [reply packages/builder/builder-doc/zh/config/tools/babel.md:7]: 那 h4 还保留不

Comment (chenjiahan) [reply packages/builder/builder-doc/zh/config/tools/babel.md:6]: h4 可以用滴，没问题

**Pull request merged**